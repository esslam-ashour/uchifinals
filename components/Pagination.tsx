"use client";

interface PaginationProps {
  totalItems: number;
  itemsPerPage: number;
  currentPage: number;
  onPageChange: (page: number) => void;
}

export default function Pagination({ totalItems, itemsPerPage, currentPage, onPageChange }: PaginationProps) {
  const totalPages = Math.ceil(totalItems / itemsPerPage);
  const maxVisiblePages = 10;
  const startPage = Math.floor((currentPage - 1) / maxVisiblePages) * maxVisiblePages + 1;
  const endPage = Math.min(startPage + maxVisiblePages - 1, totalPages);

  const handlePageChange = (page: number) => {
    onPageChange(page);
  };

  const handleNextPageGroup = () => {
    if (endPage < totalPages) onPageChange(endPage + 1);
  };

  const handlePreviousPageGroup = () => {
    if (startPage > 1) onPageChange(startPage - 1);
  };

  return (
    <div className="flex justify-center items-center space-x-2 mt-4">
      {startPage > 1 && (
        <button className="px-3 py-1 border rounded-3xl" onClick={handlePreviousPageGroup}>
          ...
        </button>
      )}
      {Array.from({ length: endPage - startPage + 1 }, (_, idx) => {
        const page = startPage + idx;
        return (
          <button
            key={page}
            className={`px-3 py-1 border rounded-3xl ${currentPage === page ? 'bg-blue-500 text-white' : ''}`}
            onClick={() => handlePageChange(page)}
          >
            {page}
          </button>
        );
      })}
      {endPage < totalPages && (
        <button className="px-3 py-1 border rounded-3xl" onClick={handleNextPageGroup}>
          ...
        </button>
      )}
    </div>
  );
}