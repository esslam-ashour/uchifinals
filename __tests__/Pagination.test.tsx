import { render } from '@testing-library/react';
import Pagination from '@/components/Pagination';

describe('Pagination', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Pagination
        totalItems={100}
        itemsPerPage={10}
        currentPage={1}
        onPageChange={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});