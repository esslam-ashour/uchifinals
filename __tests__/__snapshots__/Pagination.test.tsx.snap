// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div>
  <div
    class="jsx-a5b8bf6edb9ebf26 flex flex-wrap justify-center items-center space-x-2 mt-4"
  >
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl bg-blue-500 text-white"
    >
      1
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      2
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      3
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      4
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      5
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      6
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      7
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      8
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      9
    </button>
    <button
      class="jsx-a5b8bf6edb9ebf26 px-2 py-1 border rounded-3xl hover:bg-blue-500 hover:text-white duration-300"
    >
      10
    </button>
  </div>
</div>
`;
